packages/Kokkos/test/tstKokkosView.cpp

```diff
@@ -0,0 +1,266 @@
+// TEST FUNCTORS
+//---------------------------------------------------------------------------//
+// Fill a view with an index.
+template<class View>
+class FillFunctor
+{
+  public:
+
+    FillFunctor( View data )
+        : _data( data )
+    { /* ... */ }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const size_t i) const 
+    { _data(i) = static_cast<typename View::traits::value_type>(i); }
+
+  private:
+
+    View _data;
+};
+
+//---------------------------------------------------------------------------//
+// Assign one 2D view to another.
+template<class View1, class View2>
+class AssignFunctor
+{
+  public:
+
+    AssignFunctor( View1 view_1, View2 view_2 )
+        : _view_1(view_1)
+        , _view_2(view_2)
+    {
+        static_assert( 
+            std::is_same<typename View1::traits::value_type,
+                         typename View2::traits::value_type>::value,
+            "View data types must be the same" );
+        static_assert( 
+            std::is_same<typename View1::traits::device_type,
+                         typename View2::traits::device_type>::value,
+            "View device types must be the same" );
+        static_assert( static_cast<unsigned>(View1::Rank) == 2,
+                       "View ranks must be 2" );
+        static_assert( static_cast<unsigned>(View2::Rank) == 2,
+                       "View ranks must be 2" );
+        DTK_REQUIRE( view_1.extent(0) == view_2.extent(0) );
+        DTK_REQUIRE( view_1.extent(1) == view_2.extent(1) );
+        _extent = view_1.extent(1);
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const size_t i) const 
+    { 
+        for ( int n = 0; n < _extent; ++n ) 
+        {
+            _view_2(i,n) = _view_1(i,n);
+        }
+    }
+
+  private:
+
+    int _extent;
+    View1 _view_1;
+    View2 _view_2;
+};
+
+//---------------------------------------------------------------------------//
+// Sum the values in a view.
+template<class Scalar, class View>
+class SumFunctor
+{
+  public:
+
+    SumFunctor( View data )
+        : _data( data )
+    {
+        static_assert( 
+            std::is_same<typename View::traits::value_type,Scalar>::value,
+            "View data type must be the same as Scalar" );
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const size_t i, Scalar& val) const 
+    { val += _data(i); }
+
+  private:
+
+    View _data;
+};
+
+//---------------------------------------------------------------------------//
+// TEST TEMPLATE DECLARATIONS
+//---------------------------------------------------------------------------//
+// Test creating a view and run a basic parallel for kernel.
+TEUCHOS_UNIT_TEST_TEMPLATE_2_DECL( View, basic_for_kernel, Scalar, Node )
+{
+    // Get types.
+    using DeviceType = typename Node::device_type;
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    // Create a view in the execution space.
+    using ViewType = Kokkos::View<Scalar*,DeviceType>;
```

Comment:
Thanks, never saw that before and did not realize it was the new way. Do I understand it correctly that it can be used as a typedef, but does not actually create a new type? Does it show as the original type during debugging?